### DISCUSSION TITLE
Welcome to VDRplayer Discussions!

### DISCUSSION THREAD

**Discussion by transmitterdan** [transmitterdan/VDRplayer#2] [General]
## 👋 Welcome!
* **VDRplayer**
  VDRplayer is a handy utility for replaying NMEA0183 message files.  Typically these are files that are "recorded" during a ship's voyage.  The messages are in a special text (UTF-8) format that encode data such as GPS coordinates, depth of water, wind speed/direction, ship's heading and many other parameters.  A navigation program that accepts NMEA messages via IP protocol can receive messages from VDRplayer.  Then the navigation program "thinks" it is getting data from real navigation instruments and will act accordingly.  A popular navigation program is OpenCPN:

![OpenCPN Screen shot](https://github.com/transmitterdan/VDRplayer/blob/transmitterdan-image-1/image.png?raw=true)

  Here we see a typical scene created by sending a voyage data file from VDRplayer to OpenCPN.

  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rgleason** [comment]
[OpenCPN](https://opencpn.org/) Team Member

This is a great tool for testing OpenCPN because it runs totally separately from OpenCPN which is a big advantage when testing massive AIS nmea files because OpenCPN performance is more realistic. Some of the tests that can be run are AIS, MOB, GPS, Autoplot, verify Dashboard, Tactics etc.   To run these tests it would be great to have a good library of nmea data files that are useful for certain specific tests.  I have a very large library of these files, somewhat labeled appropriately, but not really organized.

**Reply by transmitterdan** [reply]
I accept pull requests if the NMEA files are license free or MIT style. We cannot host proprietary data. So be sure to include a license file with the PR.

**Comment by rgleason** [comment]
These are all just recordings by various users using VDR_pi or VDRplayer contributed freely with no license. What do you suggest doing?  Should I just make a nmea directory in my local fork and add a license?  Then try to clean up the files, then push?

**Comment by transmitterdan** [comment]
Just write in the commit comment that the files are freely contributed.  Create a single commit for files from each individual contributor.  That way if someone later complains they don't want their files online we can revert that commit.  Once you push the commit to your fork then create a PR from that.

**Comment by transmitterdan** [comment]
Another way to identify sources of NMEA data files is to supply a link to online files.  There are several good NMEA data files on this page: [NavMonPC](http://navmonpc.com/downloads.html)
